Core/src/TrackFinding/MeasurementSelector.cpp

```diff
@@ -64,4 +65,38 @@ double MeasurementSelector::calculateChi2(
       });
 }
 
+MeasurementSelector::Cuts MeasurementSelector::getCutsByEta(
+    const MeasurementSelectorCuts& config, double eta) {
+  const double etaAbs = std::abs(eta);
+  const auto it =
+      std::lower_bound(config.etaBins.begin(), config.etaBins.end(), etaAbs);
+  const std::size_t bin = std::distance(config.etaBins.begin(), it);
+
+  auto select = [](const auto& vec, std::size_t idx) {
+    return vec.at(std::clamp(idx, std::size_t{0}, vec.size() - 1));
+  };
+
+  const double chi2CutOffMeasurement = select(config.chi2CutOff, bin);
+  const double chi2CutOffOutlier = config.chi2CutOffOutlier.empty()
+                                       ? std::numeric_limits<double>::infinity()
+                                       : select(config.chi2CutOffOutlier, bin);
+  const std::size_t numMeasurementsCutOff =
+      select(config.numMeasurementsCutOff, bin);
+  return {chi2CutOffMeasurement, chi2CutOffOutlier, numMeasurementsCutOff};
```

Comment:
To me it would make more sense to store the cuts directly in such a struct, rather than recreating the struct for each measurement selection.

Reply:
I agree but that would be a breaking change

Reply:
this is a `private` function (at least for the moment), so I don't think a change would be breaking.

However, I think RVO ensures that this will fill the caller's `MeasurementSelector::Cuts` variable. That may be complicated when filling into a `Result<>` (I can never remember the rules, so would need to check with godbolt), but I don't think we can easily do better with a different API.